include/sound/sof/control.h

```diff
@@ -107,9 +107,13 @@ struct sof_ipc_ctrl_data {
 	/* control data - can either be appended or DMAed from host */
 	struct sof_ipc_host_buffer buffer;
 	uint32_t num_elems;	/**< in array elems or bytes for data type */
+	uint32_t total_elems;   /**< total size if message is sent in parts */
```

Comment:
Sorry, "elems" sounds like a number of some "elements" to me, "total size" sounds like a size in bytes, which one is this?

Reply:
I was just trying to go with the legacy "elem" which is in num_elems. The explanation for num_elems is "can be bytes or array elems". So as I understand it, it can be both (the struct has union of array/binary blob). If you thing total_size is better, I'm fine with that,

Reply:
@juimonen in the context of binary data, size is more intuitive than elems.